### PR TITLE
[incubator-kie-issues#1104] Migrate Junit inside DMN

### LINKED ISSUE #1104: Migrate Junit inside DMN
Migrate JUnit version to 5 inside DMN

### PR DESCRIPTION
**Pull request by gitgabrio**
Fixes https://github.com/apache/incubator-kie-issues/issues/1104

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

- for <b>pull request and downstream checks</b>  
  - Push a new commit to the PR. An empty commit would be enough.

- for a <b>full downstream build</b>
  - for <b>github actions</b> job: add the label `run_fdb`

- for <b>Jenkins PR check only</b>
  - If you are an ASF committer for KIE podling, login to Jenkins (https://ci-builds.apache.org/job/KIE/job/drools/), go to the specific PR job, and click on `Build Now` button.
</details>

### CODE REVIEW

**Review by mariofusco — APPROVED** [review]

**Review by baldimir — APPROVED** [review]

File: kie-dmn/kie-dmn-backend/pom.xml [hunk 1]
```diff
@@ -61,24 +61,22 @@
...
       <groupId>org.assertj</groupId>
       <artifactId>assertj-core</artifactId>
       <scope>test</scope>
-    </dependency>    
+    </dependency>
+		<dependency>
+			<groupId>org.junit.jupiter</groupId>
+			<artifactId>junit-jupiter</artifactId>
+			<scope>test</scope>
+		</dependency>
```
Comment (yesamer) [inline kie-dmn/kie-dmn-backend/pom.xml]: Optional: formatting

File: kie-dmn/kie-dmn-backend/src/test/java/org/kie/dmn/backend/marshalling/v1_1/UnmarshalMarshalTest.java [hunk 2]
```diff
@@ -63,99 +63,99 @@ public class UnmarshalMarshalTest {
...
     }
 
     @Test
-    public void testDish() throws Exception {
+    void dish() throws Exception {
         testRoundTrip("org/kie/dmn/backend/marshalling/v1_1/", "dish-decision.xml");
     }
 
-    @Ignore("failing to compare over a xsi:type=\"tImport\" attribute, but why content generated 'control' need to explicit it ?")
+    @Disabled("failing to compare over a xsi:type=\"tImport\" attribute, but why content generated 'control' need to explicit it ?")
```
Comment (yesamer) [inline kie-dmn/kie-dmn-backend/src/test/java/org/kie/dmn/backend/marshalling/v1_1/UnmarshalMarshalTest.java:107]: Unrelated to this change...But I'm wondering if makes sense to create a new ticket that requires to review of all the disabled tests we have and decide if they deserve a fix or if they should be removed, WDYT?

File: kie-dmn/kie-dmn-core-jsr223-jq/src/test/java/org/kie/dmn/core/jsr223/jq/BasicTest.java [hunk 3]
```diff
@@ -33,40 +33,40 @@
...
         ctx.put("Age", 1);
         ctx.put("Previous incidents?", false);
         Object result = engine.eval(" .\"Previous incidents?\" ", new SimpleBindings( ctx ));
         LOG.info("{}", result);
         assertThat(result).isEqualTo(false);
     }
-    
+
     @Test
-    public void testTest() throws ScriptException {
```
Comment (yesamer) [inline kie-dmn/kie-dmn-core-jsr223-jq/src/test/java/org/kie/dmn/core/jsr223/jq/BasicTest.java:69]: xD

File: kie-dmn/kie-dmn-core/pom.xml [hunk 4]
```diff
@@ -156,13 +156,13 @@
...
-      <groupId>junit</groupId>
-      <artifactId>junit</artifactId>
+			<groupId>org.mockito</groupId>
+			<artifactId>mockito-core</artifactId>
       <scope>test</scope>
     </dependency>
     <dependency>
       <groupId>org.mockito</groupId>
-      <artifactId>mockito-core</artifactId>
+			<artifactId>mockito-junit-jupiter</artifactId>
```
Comment (yesamer) [inline kie-dmn/kie-dmn-core/pom.xml]: Formatting

File: kie-dmn/kie-dmn-core/pom.xml [hunk 5]
```diff
@@ -156,13 +156,13 @@
       <artifactId>pmml-model</artifactId>
     </dependency>
     <dependency>
-      <groupId>junit</groupId>
-      <artifactId>junit</artifactId>
+			<groupId>org.mockito</groupId>
+			<artifactId>mockito-core</artifactId>
```
Comment (yesamer) [inline kie-dmn/kie-dmn-core/pom.xml]: Formatting

File: kie-dmn/kie-dmn-core/pom.xml [hunk 6]
```diff
@@ -202,6 +202,11 @@
       <artifactId>yasson</artifactId>
       <scope>test</scope>
     </dependency>
+		<dependency>
+			<groupId>org.junit.jupiter</groupId>
+			<artifactId>junit-jupiter</artifactId>
+			<scope>test</scope>
+		</dependency>
```
Comment (yesamer) [inline kie-dmn/kie-dmn-core/pom.xml]: Formatting

**Review by yesamer — COMMENTED** [review]
Most of the changes on the POM are not well formatted. Aside from that, LGTM!

PS. There are a couple of conflicts

**Pull request merged**